src/ManageCourses.Api/Services/EnrichmentService.cs

```diff
@@ -76,6 +76,54 @@ public UcasInstitutionEnrichmentGetModel GetInstitutionEnrichment(string instCod
             return enrichmentGetModel;
         }
 
+        /// <summary>
+        /// gets a publishable enrichment record
+        /// Looks for the published enrichment and returns it.
+        /// If the published enrichment is ull then return the ucas contact details
+        /// </summary>
+        /// <param name="instCode">institution code for the enrichment</param>
+        /// <param name="email">email of the user</param>
+        /// <returns>An enrichment object with the enrichment data. (edge case) Null if not found</returns>
+        public InstitutionEnrichmentModel GetPublishableInstitutionEnrichmentModel(string instCode, string email)
```

Comment:
That degree of duplication is not necessary.
```
public UcasInstitutionEnrichmentGetModel GetInstitutionEnrichment(string instCode, string email) => GetInstitutionEnrichmentModelImpl(instCode, email, true)

 public InstitutionEnrichmentModel GetPublishableInstitutionEnrichmentModel(string instCode, string email) => GetInstitutionEnrichmentModelImpl(instCode, email, false)

private InstitutionEnrichmentModel GetInstitutionEnrichmentModelImpl(string instCode, string email, bool includeDrafts)
        {
            ValidateUserOrg(email, instCode);
             instCode = instCode.ToUpperInvariant();
             var enrichment = _context.InstitutionEnrichments
                .Where(ie => ie.InstCode == instCode && (includeDrafts || ie.Status == EnumStatus.Published))
                .OrderByDescending(x => x.Id)
                .Include(e => e.CreatedByUser)
                .Include(e => e.UpdatedByUser)
                .FirstOrDefault();
             var enrichmentToReturn = _converter.Convert(enrichment);
            if (enrichmentToReturn != null)
            {
                return enrichmentToReturn.EnrichmentModel;
            }
            else
            {
                var ucasInst = _context.UcasInstitutions.SingleOrDefault(x => x.InstCode == instCode);
                if (ucasInst != null)
                {
                    var ucasEnrichmentModel = new InstitutionEnrichmentModel
                    {
                        Email = ucasInst.Email,
                        Telephone = ucasInst.Telephone,
                        Website = ucasInst.Url,
                        Address1 = ucasInst.Addr1,
                        Address2 = ucasInst.Addr2,
                        Address3 = ucasInst.Addr3,
                        Address4 = ucasInst.Addr4,
                        Postcode = ucasInst.Postcode,
                    };
                    return ucasEnrichmentModel;
                }                
            }
             return null;
        }
```

similarly, a PublishableCourseEnrichmentModel method is missing

reject